### PR TITLE
refactor(data-table-manager): generate readme, move conditional prop types validation to runtime checks

### PR DESCRIPTION
**Pull request by emmenko**

### CODE REVIEW

File: generators/readme/src/index.ts [hunk 1]
```diff
@@ -99,6 +99,26 @@ const parseMarkdownFragmentToAST = (fragmentContent: VFileCompatible) => {
   return fragmentAST.children;
 };
 const embeddedDefaultValueRegex = /@@defaultValue@@:\s(.*)$/;
+const parseEmbeddedDefaultValue = (
```
Comment (emmenko) [inline generators/readme/src/index.ts:102]: Just extracting the logic into a function

File: generators/readme/src/index.ts [hunk 2]
```diff
@@ -445,9 +461,12 @@ export async function transformDocument(
...
 }
 
-export async function generate(packagePath: string, flags: CommandFlags) {
+export async function generate(
+  relativePackagePath: string,
+  flags: CommandFlags
+) {
   const options: GeneratorReadmeOptions = {
-    packagePath,
+    packagePath: path.resolve(process.cwd(), relativePackagePath),
```
Comment (emmenko) [inline generators/readme/src/index.ts:469]: To be able to define the path relative to where the script is executed.

File: packages/components/data-table-manager/src/data-table-manager.js [hunk 3]
```diff
@@ -37,77 +40,202 @@ const DataTableManager = (props) => {
...
+   */
+  onSettingsChange: PropTypes.func,
+  /**
+   * A React node for rendering additional information within the table manager.
+   */
   topBar: PropTypes.node,
 };
-DataTableManager.defaultProps = {
-  isCondensed: false,
-  itemRenderer: (row, column) => row[column.key],
```
Comment (emmenko) [inline packages/components/data-table-manager/src/data-table-manager.js:104]: This was a leftover I think

File: packages/components/data-table-manager/src/data-table-manager.js [hunk 4]
```diff
@@ -37,77 +40,202 @@ const DataTableManager = (props) => {
...
+   * Signature: `(action: string, nextValue: object) => void`
+   */
+  onSettingsChange: PropTypes.func,
+  /**
+   * A React node for rendering additional information within the table manager.
+   */
   topBar: PropTypes.node,
 };
-DataTableManager.defaultProps = {
-  isCondensed: false,
```
Comment (emmenko) [inline packages/components/data-table-manager/src/data-table-manager.js:103]: This is part of the `displaySettings`

File: packages/components/data-table-manager/src/data-table-manager.js [hunk 5]
```diff
@@ -37,77 +40,202 @@ const DataTableManager = (props) => {
...
-  itemRenderer: (row, column) => row[column.key],
-  displaySettings: {
-    disableDisplaySettings: true,
-  },
-  columnManager: { disableColumnManager: true },
-  isWrappingText: false,
```
Comment (emmenko) [inline packages/components/data-table-manager/src/data-table-manager.js:109]: This is part of the `displaySettings`

File: packages/components/data-table-manager/src/data-table-manager.js [hunk 6]
```diff
@@ -37,77 +40,202 @@ const DataTableManager = (props) => {
...
+   */
   topBar: PropTypes.node,
 };
-DataTableManager.defaultProps = {
-  isCondensed: false,
-  itemRenderer: (row, column) => row[column.key],
-  displaySettings: {
-    disableDisplaySettings: true,
-  },
-  columnManager: { disableColumnManager: true },
```
Comment (emmenko) [inline packages/components/data-table-manager/src/data-table-manager.js:108]: Each `displaySettings` or `columnManager` object is in theory optional, so we shouldn't initialize it here.

File: packages/components/data-table-manager/src/data-table-settings/data-table-settings.js [hunk 7]
```diff
@@ -87,19 +103,18 @@ const DataTableSettings = (props) => {
       {openedPanelId === DISPLAY_SETTINGS && (
         <DensityManager
           data-testid={DISPLAY_SETTINGS}
-          {...props.displaySettings}
-          isDensityManagerOpenDefault={true}
+          {...(props.displaySettings || {})}
```
Comment (emmenko) [inline packages/components/data-table-manager/src/data-table-settings/data-table-settings.js:106]: Here we ensure the object exists

File: packages/components/data-table-manager/src/data-table-settings/data-table-settings.js [hunk 8]
```diff
@@ -87,19 +103,18 @@ const DataTableSettings = (props) => {
...
           onClose={handleSettingsPanelChange}
-          onDensityDisplayChange={() => {
+          onDensityDisplayChange={(event) => {
             props.onSettingsChange(
               UPDATE_ACTIONS.IS_TABLE_CONDENSED_UPDATE,
-              !props.displaySettings.isCondensed
+              event.target.value === DENSITY_COMPACT
```
Comment (emmenko) [inline packages/components/data-table-manager/src/data-table-settings/data-table-settings.js:111]: Better to read the value from the change event.

File: packages/components/data-table-manager/src/data-table-settings/data-table-settings.js [hunk 9]
```diff
@@ -108,7 +123,7 @@ const DataTableSettings = (props) => {
       {openedPanelId === COLUMN_MANAGER && (
         <ColumnManager
           data-testid={COLUMN_MANAGER}
-          {...props.columnManager}
+          {...(props.columnManager || {})}
```
Comment (emmenko) [inline packages/components/data-table-manager/src/data-table-settings/data-table-settings.js:126]: Also here we ensure the object is initialized

File: packages/components/data-table-manager/src/data-table-settings/density-manager/density-manager.js [hunk 10]
```diff
@@ -123,14 +123,18 @@ const DensityManager = (props) => {
 };
 
 DensityManager.displayName = 'DensityManager';
+DensityManager.defaultProps = {
+  isCondensed: false,
+  isWrappingText: false,
+};
```
Comment (emmenko) [inline packages/components/data-table-manager/src/data-table-settings/density-manager/density-manager.js:129]: I think it makes sense to initialize those props only here, when the density manager is rendered.

File: packages/components/data-table-manager/package.json [hunk 11]
```diff
@@ -40,7 +41,8 @@
     "lodash": "4.17.20",
     "prop-types": "15.7.2",
     "react-beautiful-dnd": "13.0.0",
-    "react-required-if": "1.0.3"
+    "react-required-if": "1.0.3",
```
Comment (jonnybel) [inline packages/components/data-table-manager/package.json:44]: Can we remove the dependency now? Or is still used somewhere else in the component?

**Review by jonnybel — APPROVED** [review]
Looks good

File: packages/components/data-table-manager/package.json (hunk 11 above)
Comment (emmenko) [reply packages/components/data-table-manager/package.json:44]: It's still used, I'll see if I can remove it when we refactor the tests, etc.

**Pull request merged**